xinference/model/embedding/core.py

```diff
@@ -595,6 +601,49 @@ def encode(
 
             return all_embeddings, all_token_nums
 
+        @no_type_check
+        def _encode_clip(
+            model: SentenceTransformer,
+            sentences: Union[Dict[str, str], List[Dict[str, str]]],
+            convert_to_numpy: bool = True,
+            **kwargs,
+        ):
+            import base64
+            import re
+            from io import BytesIO
+
+            from PIL import Image
+
+            def base64_to_image(base64_str: str) -> Image.Image:
+                # base64_data = re.sub("^data:image/.+;base64,", "", base64_str)
+                base64_data = base64_str.split(",", 1)[1]
+                byte_data = base64.b64decode(base64_data)
+                image_data = BytesIO(byte_data)
+                img = Image.open(image_data)
+                return img
+
+            all_token_nums = 0
+            all_embeddings = []
+
+            for data in sentences:
+                if data.get("text") is not None:
+                    # texts.append(data['text'])
+                    txt_embed = model.encode([data["text"]])
+                    all_embeddings.append(txt_embed)
+                    all_token_nums += len(model.tokenize(data["text"]))
+                elif data.get("image") is not None:
```

Comment:
It seems that the now we cannot get the image token nums?

Reply:
Yes. TBD. For now, I didn't find the function to calculate the image token nums.